core/trino-main/src/main/java/io/trino/execution/CreateTableTask.java

```diff
@@ -166,7 +166,7 @@ public String explain(CreateTable statement, List<Expression> parameters)
             }
             else if (element instanceof LikeClause) {
                 LikeClause likeClause = (LikeClause) element;
-                QualifiedObjectName likeTableName = createQualifiedObjectName(session, statement, likeClause.getTableName());
+                QualifiedObjectName likeTableName = metadata.redirectTable(session, createQualifiedObjectName(session, statement, likeClause.getTableName()));
```

Comment:
If we redirect here, then error messages below will be confusing for users, e.g:
```
CREATE TABLE foo LIKE source_catalog.bar
```
with error
```
LIKE table catalog `target_catalog` does not exist
```

If `getTableHandle` was redirection aware, we could run this checks on original table name and only within `getTableHandle` throw redirection specific errors.

Reply:
With the new implementation:

```
CREATE TABLE foo LIKE source_catalog.bar
```

* if source_catalog didn't exist, it's reported as is.
* if target_catalog didn't exist, it'd be reported in the error message, along with the message that the table was redirected from a source_catalog table.
* If target_catalog table is different from source_catalog, the error message `LIKE table across catalogs is not supported ...` would also contain information about redirection.

However, any issues with ACL checks would still be reported on the target. (similar to StatementAnalyzer#visitTable)